jscomp/bsb/bsb_helper_dep_graph.ml

```diff
@@ -0,0 +1,114 @@
+(* Copyright (C) 2015-2016 Bloomberg Finance L.P.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * In addition to the permissions granted to you by the LGPL, you may combine
+ * or link a "work that uses the Library" with a publicly distributed version
+ * of this file to produce a combined library or application, then distribute
+ * that combined work under the terms of your choosing, with no requirement
+ * to comply with the obligations normally placed on you by section 4 of the
+ * LGPL version 3 (or the corresponding section of a later version of the LGPL
+ * should you choose to use a later version).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
+
+let sort_files_by_dependencies ~domain dependency_graph =
+  let next current =
+    (String_map.find_exn  current dependency_graph) in
+  let worklist = ref domain in
+  let result = Queue.create () in
+  let rec visit visiting path current =
+    if String_set.mem current visiting then
+      Format.fprintf Format.err_formatter "Cyclic depends : @[%a@]"
+        (Format.pp_print_list ~pp_sep:Format.pp_print_space
+           Format.pp_print_string)
+        (current::path)
+    else if String_set.mem current !worklist then
+      begin
+        next current |>
+        String_set.iter
```

Comment:
can you mark it a duplication to some modules, so in the future, we can do refactoring to get rid of it